sklearn/tests/test_docstring_parameters.py

```diff
@@ -161,13 +167,82 @@ def test_tabs():
+               'NoSampleWeightWrapper', 'OneVsOneClassifier',
+               'OneVsRestClassifier', 'OutputCodeClassifier', 'Pipeline',
+               'RFE', 'RFECV', 'RandomizedSearchCV', 'RegressorChain',
+               'SelectFromModel', 'SparseCoder', 'SparseRandomProjection',
+               'SpectralBiclustering', 'StackingClassifier',
+               'StackingRegressor', 'TfidfVectorizer', 'VotingClassifier',
+               'VotingRegressor'}
+    if Estimator.__name__ in IGNORED or Estimator.__name__.startswith('_'):
+        pytest.skip("Estimator cannot be fit easily to test fit attributes")
+
+    est = Estimator()
+
+    if Estimator.__name__ == 'SelectKBest':
+        est.k = 2
+
+    if Estimator.__name__ == 'DummyClassifier':
+        est.strategy = "stratified"
+
+    X, y = make_classification(n_samples=20, n_features=3,
+                               n_redundant=0, n_classes=2,
+                               random_state=2)
+
+    y = _enforce_estimator_tags_y(est, y)
+    X = _enforce_estimator_tags_x(est, X)
+
+    if '1dlabels' in _safe_tags(est, 'X_types'):
+        est.fit(y)
+    elif '2dlabels' in _safe_tags(est, 'X_types'):
+        est.fit(np.c_[y, y])
+    else:
+        est.fit(X, y)
+
+    for attr in attributes:
+        desc = ' '.join(attr.desc).lower()
+        # As certain attributes are present "only" if a certain parameter is
+        # provided, this checks if the word "only" is present in the attribute
+        # description, and if not the attribute is required to be present.
+        if 'only ' not in desc:
+            assert hasattr(est, attr.name)
+
+    IGNORED = {'BayesianRidge', 'Birch', 'CCA', 'CategoricalNB', 'ElasticNet',
+               'ElasticNetCV', 'GaussianProcessClassifier',
+               'GradientBoostingRegressor', 'HistGradientBoostingClassifier',
+               'HistGradientBoostingRegressor', 'IsolationForest',
+               'KNeighborsClassifier', 'KNeighborsRegressor',
+               'KNeighborsTransformer', 'KernelCenterer', 'KernelDensity',
+               'LarsCV', 'Lasso', 'LassoLarsCV', 'LassoLarsIC',
+               'LatentDirichletAllocation', 'LocalOutlierFactor', 'MDS',
+               'MiniBatchKMeans', 'MLPClassifier', 'MLPRegressor',
+               'MultiTaskElasticNet', 'MultiTaskElasticNetCV',
+               'MultiTaskLasso', 'MultiTaskLassoCV', 'NearestNeighbors',
+               'NuSVR', 'OAS', 'OneClassSVM', 'OrthogonalMatchingPursuit',
+               'PLSCanonical', 'PLSRegression', 'PLSSVD',
+               'PassiveAggressiveClassifier', 'Perceptron', 'RBFSampler',
+               'RadiusNeighborsClassifier', 'RadiusNeighborsRegressor',
+               'RadiusNeighborsTransformer', 'RandomTreesEmbedding', 'SVR',
+               'SkewedChi2Sampler'}
+    if Estimator.__name__ in IGNORED:
+        pytest.xfail(
+            reason="Classifier has too many undocumented attributes.")
```

Comment:
FYI: we can now also put these in `_xfail_test` estimator tag for individual estimators (https://scikit-learn.org/dev/developers/develop.html#estimator-tags) but it's not critical.